runtime/parachains/src/disputes/slashing/benchmarking.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+
+use crate::{disputes::SlashingHandler, initializer, shared};
+use frame_benchmarking::{benchmarks, whitelist_account};
+use frame_support::traits::{OnFinalize, OnInitialize};
+use frame_system::RawOrigin;
+use pallet_staking::testing_utils::create_validators;
+use primitives::v2::{Hash, PARACHAIN_KEY_TYPE_ID};
+use sp_runtime::traits::{One, StaticLookup};
+use sp_session::MembershipProof;
+
+// Candidate hash of the disputed candidate.
+const CANDIDATE_HASH: CandidateHash = CandidateHash(Hash::zero());
+// Should be bumped once we support more.
+const MAX_VALIDATORS: u32 = 1 * 1024;
```

Comment:
Is this constant defined elsewere ?

Reply:
This is only used for benchmarking. I can bump it to a higher value, but it will only make generating weights slower.

Reply:
A `T::BenchmarkingConfig` is an option.

Reply:
Assuming that the weight is linear in number of validators, I'd expect that the number of validators is higher than what is benchmarked, it'd extrapolate. Isn't that the case?

If it is, then why do we need to worry this being configurable assuming that 1000 / steps items is more than enough to figure out the slope?

Reply:
introduced `T::BenchmarkingConfig`